inf-clojure.el

```diff
@@ -631,7 +639,8 @@ to continue it."
   (setq mode-line-process '(":%s"))
   (clojure-mode-variables)
   (clojure-font-lock-setup)
-  (inf-clojure-eldoc-setup)
+  (when inf-clojure-eldoc-enabledp
```

Comment:
I'm ok with this, but it might be better to make the eldoc integration a no-op instead, so that people won't have to disable/enable inf-clojure-mode for the new config to be taken into account.

Reply:
That was originally my solution, but that still leads to some problems. Eldoc is initialized as follows:

```emacs-lisp
(defun inf-clojure-eldoc-setup ()
  "Turn on eldoc mode in the current buffer."
  (setq-local eldoc-documentation-function #'inf-clojure-eldoc)
  (apply #'eldoc-add-command inf-clojure-extra-eldoc-commands))
```

Since it clobbers any existing `eldoc-documentation-function`, once eldoc is initialized in a buffer there's no good way to go back to a clean slate since any previous eldoc function is gone. I'll add the `enabledp` function to the eldoc function so it would prevent more eldocs, but it cannot restore the previous eldoc unfortunately.